fuzzy.go

```diff
@@ -96,26 +98,29 @@ func Find(pattern string, data []string) []Match {
 					matchedIndex = j
 				}
 			}
-			nextp := ""
-			if patternIndex < len(pattern)-1 {
-				nextp = strings.ToLower(string(pattern[patternIndex+1]))
+			var nextp rune
```

Comment:
You can get ~12% speedup by reusing nextp as the next candidate

Reply:
Nice! I'll leave @ericpauley to decide if they want to tackle this now. If not let's tackle this in another change :)

Thanks @lunixbochs.

Reply:
I think you may have meant `nextc`. I modified the candidate traversal to take advantage of this, comes out to a 4-6% speedup all told.

Reply:
Ah, you're right, I had a few other optimizations. The rest is here (using an interface for sorting rather than a closure, and not making a duplicate call for `bonus` (looks like it doesn't get merged): https://github.com/lunixbochs/fuzzy/commit/23dec577d3cf95de1ba7d19c921a27122c209bac

Reply:
How does it increase allocation? It's not really creating a new type, just associating methods to []Match at compile-time.. Using an interface like this even seems to reduce allocation for me (not allocating the closure).